src/utils/composables/useMovableElement.composable.js

```diff
@@ -132,9 +135,22 @@ export function useMovableElement(element, options = {}) {
     }
 
     function placeElementAt(top, left) {
-        const htmlElementStyle = toValue(element).style
-        htmlElementStyle.top = `${top}px`
-        htmlElementStyle.left = `${left}px`
+        const htmlElement = toValue(element)
+        const htmlElementStyle = htmlElement.style
+
+        // In case the original element has CSS class that affects its position, we remove them first
+        if (firstMovement && initialPositionClasses.length > 0) {
+            const rect = element.getBoundingClientRect()
+            initialPositionClasses.forEach((className) => {
+                htmlElement.classList.remove(className)
+            })
+            htmlElementStyle.top = `${rect.top}px`
+            htmlElementStyle.left = `${rect.left}px`
+        } else {
+            htmlElementStyle.top = `${top}px`
+            htmlElementStyle.left = `${left}px`
+        }
+        firstMovement = false
```

Comment:
Shouldn't you also be adding them back after the move ended?

Reply:
No, because after we move the modal, the location of the modal will be handled by the `useMovableElement`. If not, the modal will jump to another location (e.g. if there is a `translate` or `start-50` initially)

Reply:
Ok meaning you can give your modal some initial placement with for instance `end-0` (float right) and set it here in this `initialPositionClasses` so that it will be removed and placement be entirely managed by this composable 👍

Reply:
yes, exactly